Mathlib/GroupTheory/Subgroup/MulOpposite.lean

```diff
@@ -17,6 +17,52 @@ subgroup, subgroups
 
 
 variable {G : Type*} [Group G]
+variable {M : Type*} [MulOneClass M]
+
+namespace Submonoid
+
+
+/-- pull a submonoid back to an opposite submonoid along `unop`-/
+@[to_additive (attr := simps)]
+def op {M : Type*} [MulOneClass M] (x : Submonoid M) : Submonoid (MulOpposite M) where
+  carrier := MulOpposite.unop ⁻¹' x.1
+  mul_mem' {a b} ha hb := by
+    simp only [Set.mem_preimage, MulOpposite.unop_mul, SetLike.mem_coe] at ha hb ⊢
+    exact mul_mem hb ha
+  one_mem' := Submonoid.one_mem' _
+
+/-- pull an opposite submonoid back to a submonoid along `op`-/
+@[to_additive (attr := simps)]
+def unop {M : Type*} [MulOneClass M] (x : Submonoid (MulOpposite M)) : Submonoid M where
+  carrier := MulOpposite.op ⁻¹' x.1
+  mul_mem' {a b} ha hb := by
+    simp only [Set.mem_preimage, MulOpposite.unop_mul, SetLike.mem_coe] at ha hb ⊢
```

Comment:
I would guess that you can delete this line.

Reply:
The best I can do is `simpa using mul_mem ha hb`